### ISSUE TITLE
Error connecting to the service protocol: No Flutter view is available on Linux.

### ISSUE THREAD

**Issue by ansarizafar** [flutter/flutter#56268]
I am on latest Master branch after using Flutter upgrade I am getting this error

````
Error connecting to the service protocol: No Flutter view is available on Linux.
````
````
zafar@home:~/projects/dart/restock$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, 1.18.0-9.0.pre.101, on Linux, locale en_US.UTF-8)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] Chrome - develop for the web
[✓] Linux toolchain - develop for Linux desktop
[!] Android Studio (version 2.3)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[✓] VS Code (version 1.44.2)
[✓] Connected device (3 available)

! Doctor found issues in 1 category.
````

**Comment by TahaTesser** [comment]
Hi @ansarizafar 
Can you please give more details? Are you trying app on emulator or devoice? 
Is this a Flutter app or existing app with Flutter module?

can you please provide your `flutter doctor -v` and if Flutter app `flutter run --verbose`? 
Also, to better address the issue, would be helpful
if you could post a minimal code sample to reproduce the problem
Thank you

**Comment by ansarizafar** [comment]
> Are you trying app on emulator or devoice?
Is this a Flutter app or existing app with Flutter module?

I am trying to run a new app on Linux (Ubuntu). I am not using emulator. Here are the commands I am using.

```
flutter create myapp
flutter run -d Linux

```
flutter doctor -v
````
zafar@home:~/projects/dart/restock$ flutter doctor -v
[✓] Flutter (Channel master, 1.18.0-9.0.pre.98, on Linux, locale en_US.UTF-8)
    • Flutter version 1.18.0-9.0.pre.98 at /home/zafar/flutter
    • Framework revision a32c8e84bd (25 hours ago), 2020-05-03 05:54:01 -0400
    • Engine revision e3cb6812ed
    • Dart version 2.9.0 (build 2.9.0-5.0.dev c3ce873556)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /usr/local/opt/android-sdk-linux
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = /usr/local/opt/android-sdk-linux
    • Java binary at: /home/zafar/softwares/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang++ 9.0.0
    • GNU Make 4.1

[!] Android Studio (version 2.3)
    • Android Studio at /home/zafar/softwares/android-studio
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] VS Code (version 1.44.2)
    • VS Code at /usr/share/code
    • Flutter extension version 3.9.1

[✓] Connected device (3 available)
    • Linux      • Linux      • linux-x64      • Linux
    • Web Server • web-server • web-javascript • Flutter Tools
    • Chrome     • chrome     • web-javascript • Google Chrome 76.0.3809.100

! Doctor found issues in 1 category.

````
flutter run --verbose
````
zafar@home:~/projects/dart/restock$ flutter run --verbose
[  +16 ms] executing: [/home/zafar/flutter/] git -c log.showSignature=false log -n 1
--pretty=format:%H
[  +67 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] a32c8e84bda53ea763373c6d801cb9a48620c2b2
[   +2 ms] executing: [/home/zafar/flutter/] git tag --contains HEAD
[ +245 ms] Exit code 0 from: git tag --contains HEAD
[   +4 ms] executing: [/home/zafar/flutter/] git describe --match *.*.*-*.*.pre --first-parent
--long --tags
[ +189 ms] Exit code 0 from: git describe --match *.*.*-*.*.pre --first-parent --long --tags
[        ] 1.18.0-8.0.pre-98-ga32c8e8
[  +13 ms] executing: [/home/zafar/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[   +8 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/master
[        ] executing: [/home/zafar/flutter/] git ls-remote --get-url origin
[  +10 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +94 ms] executing: [/home/zafar/flutter/] git rev-parse --abbrev-ref HEAD
[  +15 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] master
[  +86 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +5 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +36 ms] executing: /usr/local/opt/android-sdk-linux/platform-tools/adb devices -l
[  +90 ms] List of devices attached
[  +12 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[  +37 ms] executing: unzip -t -qq
/home/zafar/flutter/bin/cache/downloads/storage.googleapis.com/flutter_infra/flutter/e3cb6812ed6d
a882663fc3d3bfb46464726ee61c/flutter-web-sdk-linux-x64.zip
[  +88 ms] Downloading Web SDK...
[  +19 ms] Downloading:
https://storage.googleapis.com/flutter_infra/flutter/e3cb6812ed6da882663fc3d3bfb46464726ee61c/flu
tter-web-sdk-linux-x64.zip
[+1707 ms] Received response from server, collecting bytes...
[+41474 ms] Downloading Web SDK... (completed in 43.2s)
[        ] executing: unzip -o -q
/home/zafar/flutter/bin/cache/downloads/storage.googleapis.com/flutter_infra/flutter/e3cb6812ed6d
a882663fc3d3bfb46464726ee61c/flutter-web-sdk-linux-x64.zip -d
/home/zafar/flutter/bin/cache/flutter_web_sdk
[ +588 ms] Exit code 0 from: unzip -o -q
/home/zafar/flutter/bin/cache/downloads/storage.googleapis.com/flutter_infra/flutter/e3cb6812ed6d
a882663fc3d3bfb46464726ee61c/flutter-web-sdk-linux-x64.zip -d
/home/zafar/flutter/bin/cache/flutter_web_sdk
[+1532 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[ +128 ms] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +80 ms] More than one device connected; please specify a device with the '-d <deviceId>' flag,
or use '-d
           all' to act on all devices.
[  +79 ms] Linux      • Linux      • linux-x64      • Linux
[   +1 ms] Web Server • web-server • web-javascript • Flutter Tools
[        ] Chrome     • chrome     • web-javascript • Google Chrome 76.0.3809.100
[   +5 ms] "flutter run" took 45,925ms.

#0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)
#1      RunCommand.validateCommand (package:flutter_tools/src/commands/run.dart:334:7)
<asynchronous suspension>
#2      FlutterCommand.verifyThenRunCommand
(package:flutter_tools/src/runner/flutter_command.dart:776:11)
<asynchronous suspension>
#3      FlutterCommand.run.<anonymous closure>
(package:flutter_tools/src/runner/flutter_command.dart:693:33)
<asynchronous suspension>
#4      FlutterCommand.run.<anonymous closure>
(package:flutter_tools/src/runner/flutter_command.dart)
#5      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:29)
#6      _rootRun (dart:async/zone.dart:1184:13)
#7      _CustomZone.run (dart:async/zone.dart:1077:19)
#8      _runZoned (dart:async/zone.dart:1619:10)
#9      runZoned (dart:async/zone.dart:1539:10)
#10     AppContext.run (package:flutter_tools/src/base/context.dart:149:18)
#11     FlutterCommand.run (package:flutter_tools/src/runner/flutter_command.dart:683:20)
#12     CommandRunner.runCommand (package:args/command_runner.dart:197:27)
#13     FlutterCommandRunner.runCommand.<anonymous closure>
(package:flutter_tools/src/runner/flutter_command_runner.dart:339:21)
#14     _rootRunUnary (dart:async/zone.dart:1192:38)
#15     _CustomZone.runUnary (dart:async/zone.dart:1085:19)
#16     _FutureListener.handleValue (dart:async/future_impl.dart:141:18)
#17     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:686:45)
#18     Future._propagateToListeners (dart:async/future_impl.dart:715:32)
#19     Future._completeWithValue (dart:async/future_impl.dart:526:5)
#20     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:36:15)
#21     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:303:13)
#22     FlutterVersion.checkFlutterVersionFreshness (package:flutter_tools/src/version.dart)
#23     _rootRunUnary (dart:async/zone.dart:1192:38)
#24     _CustomZone.runUnary (dart:async/zone.dart:1085:19)
#25     _FutureListener.handleValue (dart:async/future_impl.dart:141:18)
#26     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:686:45)
#27     Future._propagateToListeners (dart:async/future_impl.dart:715:32)
#28     Future._completeWithValue (dart:async/future_impl.dart:526:5)
#29     Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:560:7)
#30     _rootRun (dart:async/zone.dart:1184:13)
#31     _CustomZone.run (dart:async/zone.dart:1077:19)
#32     _CustomZone.runGuarded (dart:async/zone.dart:979:7)
#33     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1019:23)
#34     _microtaskLoop (dart:async/schedule_microtask.dart:43:21)
#35     _startMicrotaskLoop (dart:async/schedule_microtask.dart:52:5)
#36     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
#37     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)


````

**Comment by ansarizafar** [comment]
I think the issue is related to hot reload as the app runs without error If I use --no-hot option. I don't want to use emulator on my machine, A quick fix is badly needed.

**Comment by jmagman** [comment]
@jonahwilliams you think this could be fixed by https://github.com/flutter/flutter/pull/56387 and friends?

**Comment by jonahwilliams** [comment]
Yes, this should be fixed as of 9b7b9d795edfdd12c6ad8dd6495bed789c1bac05. @ansarizafar it would be super helpful if you could checkout this version and verify.

**Comment by winksaville** [comment]
An FYI, I used `flutter run -d pixel` to test an app on my Pixel 3a and ran into a similar problem:
```
$ flutter run -d pixel
Launching lib/main.dart on Pixel 3a in debug mode...
Running Gradle task 'assembleDebug'...                                  
Running Gradle task 'assembleDebug'... Done                         3.9s
✓ Built build/app/outputs/flutter-apk/app-debug.apk.
Installing build/app/outputs/flutter-apk/app.apk...                 7.2s
Error connecting to the service protocol: No Flutter view is available on Pixel 3a.
```
I pulled in the latest master branch and retested and it worked:
```
$ flutter run -d pixel
Launching lib/main.dart on Pixel 3a in debug mode...
Running Gradle task 'assembleDebug'...                                  
Running Gradle task 'assembleDebug'... Done                         2.6s
✓ Built build/app/outputs/flutter-apk/app-debug.apk.
Waiting for Pixel 3a to report its views...                         10ms
Syncing files to device Pixel 3a...                                291ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h Repeat this help message.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).
An Observatory debugger and profiler on Pixel 3a is available at: http://127.0.0.1:33659/AiRaEubv-hE=/
```

**Comment by jonahwilliams** [comment]
Thanks @winksaville !

**Comment by ansarizafar** [comment]
@jonahwilliams I confirm its working now. Thanks for the quick fix.

**Comment by jonahwilliams** [comment]
Thanks all, sorry for the trouble!

Fixed by  #56387
